pkg/util/kubernetes/apiserver/hpa_controller.go

```diff
@@ -298,13 +316,23 @@ func (h *AutoscalersController) syncAutoscalers(key interface{}) error {
 			return ErrIsEmpty
 		}
 		new := h.hpaProc.ProcessHPAs(hpa)
+		if len(new)+h.metricsProcessedCount > maxMetricsCount {
```

Comment:
Unfortunately we can't as is, because the error is in the body of the response from the backend, but as we use the go-datadog client, for a Metrics call it returns Series which will just be empty with no error (as the Status of the response is 200).
The only way to fix that would be to change the code of the upstream library to check for empty series and error in body.
But nothing can be done in the DCA.